org.eclipse.scanning.points/src/org/eclipse/scanning/jython/JythonInterpreterManager.java

```diff
@@ -0,0 +1,243 @@
+package org.eclipse.scanning.jython;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Properties;
+
+import org.eclipse.core.runtime.FileLocator;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.scanning.points.ScanPointGeneratorFactory;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.wiring.BundleWiring;
+import org.python.core.Py;
+import org.python.core.PyString;
+import org.python.core.PySystemState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * 
+ * An interpreter manager which is designed to create an interpreter
+ * which either is sympathic and working with the GDA configuration or
+ * creates an empty and working interpreter.
+ * 
+ * @author Matthew Gerring
+ *
+ */
+class JythonInterpreterManager {
+
+	private static final Logger logger = LoggerFactory.getLogger(JythonInterpreterManager.class);
+	
+	
+	private static volatile PySystemState configuredState;
+
+	/**
+	 * Call to ensure that an interpreter is set up and configured and
+	 * able to load the relevant bundles.
+	 */
+	public static synchronized void setupSystemState(String... bundleNames) {
+		
+		ClassLoader loader=null;
+		if (configuredState==null) { // Relies on setupSystemState() being called early in the server startup.
+			loader = createJythonClassLoader(PySystemState.class.getClassLoader());
+	 		initializePythonPath(loader); 
+		}
+		
+    	PySystemState state = Py.getSystemState();
+    	if (state==configuredState) return;
+    	
+    	if (loader==null) {
+    		// Then someone else has changed the PySystemState
+    		// They will not have added our 
+			loader = createJythonClassLoader(state.getClassLoader());   // Don't clobber their working.		
+    	}
+	   	state.setClassLoader(loader);
+    	
+     	setJythonPaths(state);       // Tries to ensure that enough of jython is on the path
+     	setSpgGeneratorPaths(state, bundleNames); // Adds the scripts directory from points
+	   	Py.setSystemState(state);
+ 
+	   	configuredState = state;
+	}
+
+	private static void setSpgGeneratorPaths(PySystemState state, String... bundleNames) {
+		
+        File location = getBundleLocation("org.eclipse.scanning.points");
+        state.path.add(new PyString(location.getAbsolutePath() + "/scripts/"));
+        
+        location = getBundleLocation("org.eclipse.scanning.sequencer");
+        state.path.add(new PyString(location.getAbsolutePath() + "/scripts/"));
+
+        if (bundleNames!=null) {
+        	for (String bundle : bundleNames) {
+                location = getBundleLocation(bundle);
+                state.path.add(new PyString(location.getAbsolutePath() + "/scripts/"));
+			}
+        }
+	}
+
+
+	private static void initializePythonPath(ClassLoader loader) {
+		try {
+	    	String jythonBundleName = System.getProperty("org.eclipse.scanning.jython.osgi.bundle.name", "uk.ac.diamond.jython");
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "org.eclipse.scanning.jython.osgi.bundle.name" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "uk.ac.diamond.jython" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)